### DISCUSSION TITLE
Roadmap / timeline

### DISCUSSION THREAD

**Discussion by mnmr** [nats-io/nats.net#125] [General]
I was wondering if there is a non-binding roadmap for this project somewhere? Is it a purely community-driven thing or are there full-time developers assigned to advancing this v2 library?

We are currently looking for something to replace 0mq/NetMQ in our solution and would rather like if the answer wasn’t a huge kubernetes cluster to run Pulsar :)

**Comment by mtmk** [comment]
Our current priorities are JetStream support then KV and Object Store. We are also keen to support .NET 8 especially Native AOT. We'd love to see NATS used in the .NET ecosystem more, for example as SignalR backend similar to Redis. These could be community projects or supported by Synadia partially or fully which we'll have to decide down the line.

I am the maintainer of NATS .NET v2 project and a [Synadia](https://www.synadia.com/) software engineer. I'm working on this project full time with @caleblloyd who dedicates fair amount of time as well. I am a member of a dedicated client libraries team at Synadia.

Having said that NATS has a great community and as the .NET v2 project our community is starting to grow too. We've already accepted a few PRs for example and would love you to get involved as well. Please also do jump on [NATS Slack](https://slack.nats.io/) and join the dotnet channel.

Also to keep your mind at rest [Synadia](https://www.synadia.com/) is the company behind NATS and as well as supporting our community we do offer enterprise support too in case you need it.

You should already be able to try NATS Core features to replace 0mq using our preview release of [NATS.Client.Core](https://www.nuget.org/packages/NATS.Client.Core/2.0.0-alpha.1) ([docs here](https://nats-io.github.io/nats.net.v2/documentation/intro.html)). I'm currently working on the first preview release of JetStream support which should be released very soon.

Good luck with your project :)

**Reply by mnmr** [reply]
Thank you very much, this was exactly the kind of response I was hoping for. I will be diving into NATS to learn how it works and how well it performs for the tasks we have in mind for it (which is a fair bit of everything, but of course starting with pure messaging, as that is all we have today with 0mq).

I'm hoping that NATS has a "memory Jetstream store" or other mechanism to make messages available for a period of time (without actually writing them to a store), as we have some topics that would benefit from "some level of retention" but also cannot tolerate much latency. But I'm getting ahead of myself here - haven't even properly started reading the docs yet :)

Again, thank you - and we'll likely be trying the early alpha JS code soon.

**Comment by mikhail-barg** [comment]
@mtmk just to highlight how we appreciate your work — we are currently considering switching from RMQ to NATS, and lack of modern async .Net client is a show-stopper for us. So we really fond of your efforts and going to be early adopters of .NET v2 client.
